pkg/operator/ceph/csi/controller.go

```diff
@@ -263,3 +273,62 @@ func (r *ReconcileCSI) reconcile(request reconcile.Request) (reconcile.Result, e
 
 	return reconcileResult, nil
 }
+
+func reportNodeTelemetry(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo) {
+	// In the corner case that reconciles are started in quick succession and the telemetry
+	// hasn't had a chance to complete yet from a previous reconcile, simply allow
+	// a single goroutine to report telemetry at a time.
+	telemetryMutex.Lock()
+	defer telemetryMutex.Unlock()
+
+	logger.Info("reporting node telemetry")
+
+	// Report the K8sNodeCount
+	nodelist, err := context.Clientset.CoreV1().Nodes().List(clusterInfo.Context, metav1.ListOptions{})
+	if err != nil {
+		logger.Warningf("failed to report the K8s node count. %v", err)
+	} else {
+		telemetry.ReportKeyValue(context, clusterInfo, telemetry.K8sNodeCount, strconv.Itoa(len(nodelist.Items)))
+	}
+
+	// Report the cephNodeCount
+	// abc := nodedaemon.CrashCollectorAppName
+	listoption := metav1.ListOptions{LabelSelector: fmt.Sprintf("%q=%q", k8sutil.AppAttr, "rook-ceph-crashcollector")}
+	cephNodeList, err := context.Clientset.CoreV1().Nodes().List(clusterInfo.Context, listoption)
```

Comment:
Instead of a separate call to list the nodes multiple times, just use the same `nodeList` that was retrieved above on line 287. You can iterate over it to evaluate the labels locally, so no separate query is necessary. One query of nodes is much more efficient than five.

Reply:
True

Reply:
Have you tested this? Actually I don't think this will work. The nodes don't have labels on them for the daemons, so we likely need a separate query for each daemon to count how many pods running of each type.

Reply:
Yes, we need to revert it, doesn't work,

There is one more problem how to disable for first reconcile of cluster creation as 
```
2023-09-07 14:22:12.502209 W | telemetry: failed to set telemetry key "rook/node/count/kubernetes-total". failed to set "rook/node/count/kubernetes-total" in the mon config-key store. output: Error initializing cluster client: ObjectNotFound('RADOS object not found (error calling conf_read_file)',): exit status 1

```

Logs this